lib/ansible/modules/cloud/vmware/vmware_host_scanhba.py

```diff
@@ -0,0 +1,108 @@
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_host_scanhba
+short_description: Rescan host HBA's and optionally refresh the storage system
+description:
+- This module can force a rescan of the hosts HBA subsystem which is needed when wanting to mount a new datastore.
+- You could use this before using vmware_host_datastore to mount a new datastore to ensure your device/volume is ready.
+- You can also optionally force a Refresh of the Storage System in vCenter/ESXi Web Client.
+- All parameters and VMware object names are case sensitive.
+version_added: '2.8'
+author:
+- Michael Eaton (@if-meaton)
+notes:
+- Tested on vSphere 6.0
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  esxi_hostname:
+    description:
+    - ESXi hostname to Rescan the storage subsystem on.
+    required: true
+  refresh_storage:
+    description:
+    - Refresh the storage system in vCenter/ESXi Web Client
+    required: false
+    default: false
+    type: bool
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Recan HBA's for a given ESXi host and refresh storage system objects
+  vmware_host_scanhba:
+      hostname: '{{ vcenter_hostname }}'
+      username: '{{ vcenter_username }}'
+      password: '{{ vcenter_password }}'
+      esxi_hostname: '{{ inventory_hostname }}'
+      refresh_storage: true
+  delegate_to: localhost
+- name: Recan HBA's for a given ESXi host and don't refresh storage system objects
+  vmware_host_scanhba:
+      hostname: '{{ vcenter_hostname }}'
+      username: '{{ vcenter_username }}'
+      password: '{{ vcenter_password }}'
+      esxi_hostname: '{{ inventory_hostname }}'
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi, find_obj
+from ansible.module_utils._text import to_native
+
+
+class VmwareHbaScan(PyVmomi):
+    def __init__(self, module):
+        super(VmwareHbaScan, self).__init__(module)
+
+    def scan(self):
+        esxi_host_name = self.params.get('esxi_hostname', None)
+        refresh_storage = self.params.get('refresh_storage', bool)
+        host = find_obj(self.content, [vim.HostSystem], name=esxi_host_name)
+
+        host.configManager.storageSystem.RescanAllHba()
+        if refresh_storage is True:
+            host.configManager.storageSystem.RefreshStorageSystem()
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        esxi_hostname=dict(type='str', required=True),
+        refresh_storage=dict(type='bool', default=False, required=False)
+    )
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=False)
```

Comment:
I would recommend to add `check_mode` for this module. You can check other modules in VMware space to see how it works for them.

Reply:
What would you expect it to do in check mode? It either rescans the hbas or it doesn't.. can you elaborate on this?

Reply:
I've done as requested.